pycsw/ogc/csw/csw3.py

```diff
@@ -985,15 +985,18 @@ def getrecords(self):
                 catalogue: %s', fedcat)
                 try:
                     start_time = time()
-                    remotecsw = CatalogueServiceWeb(fedcat, skip_caps=True)
-                    remotecsw.getrecords2(xml=self.parent.request,
-                                          esn=self.parent.kvp['elementsetname'],
-                                          outputschema=self.parent.kvp['outputschema'])
+
+                    remotecsw = CatalogueServiceWeb(fedcat, version='3.0.0', skip_caps=True)
+                    if self.parent.request.startswith('http'):
+                        self.parent.request = self.parent.request.split('?')[-1]
+                    remotecsw.getrecords(xml=self.parent.request,
```

Comment:
Shouldn't this be getrecords2 as well?

Reply:
It should not.  `getrecords2` has long been planned to replace `getrecords`.

Reply:
thanks